[Ops] MWDI is stopped unexpectedly.

NP is enabled and MWDI is subscribed only for "notify-controller-attribute-value-changes".  Device disconnected and connected status are updated MWDI successfully.
Unexpectedly MWDI stopped with below error prints

Error prints:
-------------
```
[27.04.2024 00:08.33.375] [ERROR] Elasticsearch error occurred: {"_index":"mwdi_preprod_1.1.2.a_impl-000001","_type":"_doc","_id":"306558261B","_version":3,"result":"not_found","_shards":{"total":2,"successful":1,"failed":0},"_seq_no":74634,"_primary_term":1}
/home/openbackhaul/microWaveDeviceInventory/node_modules/@elastic/elasticsearch/lib/Transport.js:367
        const error = new ResponseError(result)
                      ^

ResponseError: {"_index":"mwdi_preprod_1.1.2.a_impl-000001","_type":"_doc","_id":"306558261B","_version":3,"result":"not_found","_shards":{"total":2,"successful":1,"failed":0},"_seq_no":74634,"_primary_term":1}
    at onBody (/home/openbackhaul/microWaveDeviceInventory/node_modules/@elastic/elasticsearch/lib/Transport.js:367:23)
    at IncomingMessage.onEnd (/home/openbackhaul/microWaveDeviceInventory/node_modules/@elastic/elasticsearch/lib/Transport.js:291:11)
    at IncomingMessage.emit (node:events:525:35)
    at endReadableNT (node:internal/streams/readable:1358:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  meta: {
    body: {
      _index: 'mwdi_preprod_1.1.2.a_impl-000001',
      _type: '_doc',
      _id: '306558261B',
      _version: 3,
      result: 'not_found',
      _shards: { total: 2, successful: 1, failed: 0 },
      _seq_no: 74634,
      _primary_term: 1
    },
    statusCode: 404,
    headers: {
      'x-elastic-product': 'Elasticsearch',
      warning: '299 Elasticsearch-7.14.2-6bc13727ce758c0e943c3c21653b3da82f627f75 "Elasticsearch built-in security features are not enabled. Without authentication, your cluster could be accessible to anyone. See https://www.elastic.co/guide/en/elasticsearch/reference/7.14/security-minimal-setup.html to enable security."',
      'content-type': 'application/json; charset=UTF-8',
      'content-length': '195'
    },
    meta: {
      context: null,
      request: {
        params: {
          method: 'DELETE',
          path: '/mwdi_preprod_1.1.2.a_impl/_doc/306558261B',
          body: '',
          querystring: '',
          headers: {
            'user-agent': 'elasticsearch-js/7.17.12 (linux 4.18.0-240.22.1.el8_3.x86_64-x64; Node.js v16.20.2)',
            'x-elastic-client-meta': 'es=7.17.12,js=16.20.2,t=7.17.12,hc=16.20.2'
          },
          timeout: 2000
        },
        options: {},
        id: 54805
      },
      name: 'elasticsearch-js',
      connection: Connection {
        url: <ref *1> URL {
          [Symbol(context)]: URLContext {
            flags: 400,
            scheme: 'http:',
            username: '',
            password: '',
            host: '10.20.64.4',
            port: 9200,
            path: [Array],
            query: null,
            fragment: null
          },
          [Symbol(query)]: URLSearchParams {
            [Symbol(query)]: [],
            [Symbol(context)]: [Circular *1]
          }
        },
        ssl: null,
        id: 'http://x.x.x.x:xxxx/',
        headers: { authorization: 'ApiKey API key not yet defined.' },
        deadCount: 0,
        resurrectTimeout: 0,
        caFingerprint: null,
        _openRequests: 0,
        _status: 'alive',
        roles: { master: true, data: true, ingest: true, ml: false },
        agent: Agent {
          _events: [Object: null prototype] {
            free: [Function (anonymous)],
            newListener: [Function: maybeEnableKeylog]
          },
          _eventsCount: 2,
          _maxListeners: undefined,
          defaultPort: 80,
          protocol: 'http:',
          options: [Object: null prototype] {
            keepAlive: true,
            keepAliveMsecs: 1000,
            maxSockets: 256,
            maxFreeSockets: 256,
            scheduling: 'lifo',
            path: null
          },
          requests: [Object: null prototype] {},
          sockets: [Object: null prototype] {},
          freeSockets: [Object: null prototype] { 'x.x.x.x:xxxx:': [Array] },
          keepAliveMsecs: 1000,
          keepAlive: true,
          maxSockets: 256,
          maxFreeSockets: 256,
          scheduling: 'lifo',
          maxTotalSockets: Infinity,
          totalSocketCount: 2,
          [Symbol(kCapture)]: false
        },
        makeRequest: [Function: request]
      },
      attempts: 1,
      aborted: false
    }
  }
}
```

--- a/server/service/individualServices/CyclicProcessService/cyclicProcess.js
+++ b/server/service/individualServices/CyclicProcessService/cyclicProcess.js
@@ -592,8 +592,12 @@ module.exports.deviceListSynchronization = async function deviceListSynchronizat
     //
     for (let i = 0; i < dropEsElements.length; i++) {
         let cc_id = dropEsElements[i]['node-id'];
-        let ret = await individualServicesService.deleteRecordFromElasticsearch(7, '_doc', cc_id);
-        printLog('* ' + ret.result, print_log_level >= 2);
+        try {
+            let ret = await individualServicesService.deleteRecordFromElasticsearch(7, '_doc', cc_id);
+            printLog('* ' + ret.result, print_log_level >= 2);
+        } catch(error) {
+            console.log('* An error has occurred deleting ' + cc_id + ' from elasticsearch. Element does not exist.)', print_log_level >= 2);
+        }
     }
     console.log('* ES Device List size: ' + esDeviceList.length + '                                                                             *');
     console.log('* ODL Device List size: ' + odlDeviceList.length + '                                                                            *');
@@ -662,114 +666,114 @@ module.exports.startCyclicProcess = async function startCyclicProcess(logging_le
         return;
     }
     odlDeviceList = filterConnectedDevices(odlDeviceList);
-    //printLog(printList('Device List (ODL)', odlDeviceList), print_log_level >= 1);
-
+    
+    let elasticsearchList;
     try {
-        let elasticsearchList = await individualServicesService.readDeviceListFromElasticsearch();
-        //printLog(printList('Device List (ES)', elasticsearchList), print_log_level >= 1);
-        //
-        // Calculate common elements and drop elements of ES-DL and print the ES-DL
-        //
-        let commonEsElements = [];
-        let dropEsElements = [];
-        let esRules = [];
-        let esString = 'Device List (ES): [';
-        for (let i = 0; i < elasticsearchList.length; i++) {
-            let found = false;
-            for (let j = 0; j < odlDeviceList.length; j++) {
-                if (elasticsearchList[i]['node-id'] == odlDeviceList[j]['node-id']) {
-                    found = true;
-                    commonEsElements.push(elasticsearchList[i]);
-                    esString += (((i == 0) ? '' : '|') + ('%c' + elasticsearchList[i]['node-id'] + '%c'));
-                    esRules.push("color:green");
-                    esRules.push("color:inherit");
-                    break;
-                }
-            }
-            if (!found) {
-                dropEsElements.push(elasticsearchList[i]);
+        elasticsearchList = await individualServicesService.readDeviceListFromElasticsearch();
+    } catch(error) {
+        console.log(error);
+        return;
+    }
+    //printLog(printList('Device List (ES)', elasticsearchList), print_log_level >= 1);
+    //
+    // Calculate common elements and drop elements of ES-DL and print the ES-DL
+    //
+    let commonEsElements = [];
+    let dropEsElements = [];
+    let esRules = [];
+    let esString = 'Device List (ES): [';
+    for (let i = 0; i < elasticsearchList.length; i++) {
+        let found = false;
+        for (let j = 0; j < odlDeviceList.length; j++) {
+            if (elasticsearchList[i]['node-id'] == odlDeviceList[j]['node-id']) {
+                found = true;
+                commonEsElements.push(elasticsearchList[i]);
                 esString += (((i == 0) ? '' : '|') + ('%c' + elasticsearchList[i]['node-id'] + '%c'));
-                esRules.push("color:red");
+                esRules.push("color:green");
                 esRules.push("color:inherit");
+                break;
             }
         }
-        esString += (']  (' + elasticsearchList.length + ')');
-        console.log(esString, ...esRules);
+        if (!found) {
+            dropEsElements.push(elasticsearchList[i]);
+            esString += (((i == 0) ? '' : '|') + ('%c' + elasticsearchList[i]['node-id'] + '%c'));
+            esRules.push("color:red");
+            esRules.push("color:inherit");
+        }
+    }
+    esString += (']  (' + elasticsearchList.length + ')');
+    console.log(esString, ...esRules);
 
-        //
-        // Calculate new elements and common elements of ODL-DL and print the ODL-DL
-        //
-        let newOdlElements = [];
-        let commonOdlElements = [];
-        let odlRules = [];
-        let odlString = 'Device List (ODL): [';
-        for (let i = 0; i < odlDeviceList.length; i++) {
-            let found = false;
-            for (let j = 0; j < elasticsearchList.length; j++) {
-                if (odlDeviceList[i]['node-id'] == elasticsearchList[j]['node-id']) {
-                    commonOdlElements.push(odlDeviceList[i]);
-                    odlString += (((i == 0) ? '' : '|') + ('%c' + odlDeviceList[i]['node-id'] + '%c'));
-                    odlRules.push("color:green");
-                    odlRules.push("color:inherit");
-                    found = true;
-                    break;
-                }
-            }
-            if (!found) {
-                newOdlElements.push(odlDeviceList[i])
+    //
+    // Calculate new elements and common elements of ODL-DL and print the ODL-DL
+    //
+    let newOdlElements = [];
+    let commonOdlElements = [];
+    let odlRules = [];
+    let odlString = 'Device List (ODL): [';
+    for (let i = 0; i < odlDeviceList.length; i++) {
+        let found = false;
+        for (let j = 0; j < elasticsearchList.length; j++) {
+            if (odlDeviceList[i]['node-id'] == elasticsearchList[j]['node-id']) {
+                commonOdlElements.push(odlDeviceList[i]);
                 odlString += (((i == 0) ? '' : '|') + ('%c' + odlDeviceList[i]['node-id'] + '%c'));
-                odlRules.push("color:yellow");
+                odlRules.push("color:green");
                 odlRules.push("color:inherit");
+                found = true;
+                break;
             }
         }
-        odlString += (']  (' + odlDeviceList.length + ')');
-        console.log(odlString, ...odlRules);
-        //
-        // Shuffle the new elements (commented issue 757)
-        //
-        //newOdlElements = shuffleArray(newOdlElements);
-
-        //
-        // Calculate the new ODL-DL: [new odl elements shuffled] + [common es elements]
-        //
-        let newRules = [];
-        let newString = 'Device List (Updated): [';
-        let newPiped = false;
-        for (let i = 0; i < newOdlElements.length; i++) {
-            newString += (((i == 0) ? '' : '|') + ('%c' + newOdlElements[i]['node-id'] + '%c'));
-            newRules.push("color:yellow");
-            newRules.push("color:inherit");
-            newPiped = true;
+        if (!found) {
+            newOdlElements.push(odlDeviceList[i])
+            odlString += (((i == 0) ? '' : '|') + ('%c' + odlDeviceList[i]['node-id'] + '%c'));
+            odlRules.push("color:yellow");
+            odlRules.push("color:inherit");
         }
-        for (let i = 0; i < commonEsElements.length; i++) {
-            newString += (((i == 0 && newPiped == false) ? '' : '|') + ('%c' + commonEsElements[i]['node-id'] + '%c'));
-            newRules.push("color:green");
-            newRules.push("color:inherit");
-            newPiped = true;
-        }
-        newString += (']  (' + odlDeviceList.length + ')');
-        console.log(newString, ...newRules);
+    }
+    odlString += (']  (' + odlDeviceList.length + ')');
+    console.log(odlString, ...odlRules);
+    //
+    // Shuffle the new elements (commented issue 757)
+    //
+    //newOdlElements = shuffleArray(newOdlElements);
 
-        //
-        // Drop from ES all the elements not more present in ES-DL
-        //
-        for (let i = 0; i < dropEsElements.length; i++) {
-            let cc_id = dropEsElements[i]['node-id'];
+    //
+    // Calculate the new ODL-DL: [new odl elements shuffled] + [common es elements]
+    //
+    let newRules = [];
+    let newString = 'Device List (Updated): [';
+    let newPiped = false;
+    for (let i = 0; i < newOdlElements.length; i++) {
+        newString += (((i == 0) ? '' : '|') + ('%c' + newOdlElements[i]['node-id'] + '%c'));
+        newRules.push("color:yellow");
+        newRules.push("color:inherit");
+        newPiped = true;
+    }
+    for (let i = 0; i < commonEsElements.length; i++) {
+        newString += (((i == 0 && newPiped == false) ? '' : '|') + ('%c' + commonEsElements[i]['node-id'] + '%c'));
+        newRules.push("color:green");
+        newRules.push("color:inherit");
+        newPiped = true;
+    }
+    newString += (']  (' + odlDeviceList.length + ')');
+    console.log(newString, ...newRules);
+
+    //
+    // Drop from ES all the elements not more present in ES-DL
+    //
+    for (let i = 0; i < dropEsElements.length; i++) {
+        let cc_id = dropEsElements[i]['node-id'];
+        try {
             let ret = await individualServicesService.deleteRecordFromElasticsearch(7, '_doc', cc_id);
             printLog(ret.result, print_log_level >= 2);
+        } catch(error) {
+            console.log('* An error has occurred deleting ' + cc_id + ' from elasticsearch. Element does not exist.)', print_log_level >= 2);
         }
-
-        odlDeviceList = [].concat(newOdlElements, commonEsElements);
-        printLog("Update device list to elasticsearch", print_log_level >= 2)
-    } catch (error) {
-        console.log(error);
-        // (commented issue 757)
-        // odlDeviceList = shuffleArray(odlDeviceList);
-        printLog(printList('Device List', odlDeviceList), print_log_level >= 1);
-        // printLog("Write ODL device list shuffled to Elasticsearch", print_log_level >= 2);
-        printLog("Write ODL device list to Elasticsearch", print_log_level >= 2);
     }
 
+    odlDeviceList = [].concat(newOdlElements, commonEsElements);
+    printLog("Update device list to elasticsearch", print_log_level >= 2)
+    
     let deviceListCleaned = [];
     for (let i = 0; i < odlDeviceList.length; i++) {
         deviceListCleaned.push({"node-id" : odlDeviceList[i]["node-id"]});

